src/components/QuestDetailsWrapper/index.tsx

```diff
@@ -587,14 +592,21 @@ export function QuestDetailsWrapper({
           eligible: false,
           steamAccountLinked: true
         },
-        playStreak: getPlaystreakArgsFromQuestData(
+        playStreak: getPlaystreakArgsFromQuestData({
           questMeta,
           questPlayStreakData,
-          isSignedIn
-        )
+          useModuleInitTimeForSessionStartTime: isSignedIn,
+          onSync: async() => {
+            if (questsWithExternalPlayStreakSync.includes(questMeta.id)) {
+              syncPlayStreakWithExternalSource(questMeta.id)
+            } else {
+              await syncPlaySession(projectId, 'hyperplay')
+              questPlayStreakResult.invalidateQuery()
+            }
+          }
+        })
```

Comment:
I like that the logic of when to execute an external sync is here. The drawback is the need of `questsWithExternalPlayStreakSync` but I think that's ok